packages/cozy-client/src/models/contact.js

```diff
@@ -135,13 +143,6 @@ export const getDisplayName = contact =>
  * @returns {string} - the contact's 'byFamilyNameGivenNameEmailCozyUrl' index
  */
 export const getIndexByFamilyNameGivenNameEmailCozyUrl = contact => {
-  if (get(contact, 'indexes.byFamilyNameGivenNameEmailCozyUrl')) {
-    if (isEmpty(contact.indexes.byFamilyNameGivenNameEmailCozyUrl)) {
-      return null
-    }
-    return contact.indexes.byFamilyNameGivenNameEmailCozyUrl
-  }
-
```

Comment:
So now we can have 

```
getIndexByFamilyNameGivenNameEmailCozyUrl({
  indexes: {
    byFamilyNameGivenNameEmailCozyUrl: 'toto'
  },
  name: {
     familyName: 'tutu'
   }
})
```

and this will return 'tutu' instead of 'toto' ?

It seems a bit surprising because it's not obvious from the name of the function that we're a computing a value, not just getting it.

Should we have a get and a compute function ? BTW I find the "getIndexByFamilyNameGivenNameEmailCozyUrl" name not very readable. The fact that the default index for sorting contacts uses all those fields is an implementation detail that is leaked to the caller when writing the name of the method. I think getDefaultSortIndexValue / computeDefaultSortIndexValue would be shorter and clearer.

Reply:
> It seems a bit surprising because it's not obvious from the name of the function that we're a computing a value, not just getting it.

We arrived at the same conclusion this morning during our meeting https://mattermost.cozycloud.cc/cozycloud/pl/59z41kj8ridhtjkkw4zp3t7hcy 👍

Reply:
Oui je vais découper les méthodes pour avoir le get et le make séparément

Reply:
Ça a été fait et on va pas revenir dessus, mais pour le 
"getDefaultSortIndexValue" je suis pas convaincu. On est dans une méthode de cozy-client qui est partagé à l'ensemble des app / connecteurs. Cet index de trie c'est celui par défaut pour l'application contacts, pas pour le reste des applications. Donc ça me paraît pas forcément pertinent de l'appeler defaultSortIndexValue dans ce contexte, et je préférai largement savoir sur quoi je faisais le sort que de ne rien savoir du tout.